server/client_server.go

```diff
@@ -22,10 +30,117 @@ func newServerClient(jiraClient *jira.Client) Client {
 	}
 }
 
-// GetCreateMeta returns the metadata needed to implement the UI and validation of
+type ProjectIssueInfo struct {
+	Values []*jira.MetaIssueType `json:"values"`
+}
+
+type FieldInfo struct {
+	Values []map[string]interface{} `json:"values"`
+}
+
+type JiraServerVersionInfo struct {
+	Version string `json:"version"`
+}
+
+// GetIssueInfo returns the issues information based on project id.
+func (client jiraServerClient) GetIssueInfo(projectID string) (*ProjectIssueInfo, *jira.Response, error) {
+	apiEndpoint := fmt.Sprintf("%s%s/issuetypes", APIEndpointCreateIssueMeta, projectID)
+	req, err := client.Jira.NewRequest(http.MethodGet, apiEndpoint, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	issues := ProjectIssueInfo{}
+	response, err := client.Jira.Do(req, &issues)
+	return &issues, response, err
+}
+
+func (client jiraServerClient) GetProjectInfoForPivotJiraVersion(options *jira.GetQueryOptions) (*jira.CreateMetaInfo, *jira.Response, error) {
+	var issueInfo *ProjectIssueInfo
+	var req *http.Request
+
+	projectList, resp, err := client.Jira.Project.ListWithOptions(options)
+	meta := new(jira.CreateMetaInfo)
+
+	if err == nil {
+		for _, proj := range *projectList {
```

Comment:
We can remove a level of indentation here by returning early

```suggestion
	if err != nil {
		return, nil, resp, errors.Wrap(err, "failed to list projects")
	}
	
	for _, proj := range *projectList {
```